code/datums/components/plumbing/_plumbing.dm

```diff
@@ -27,12 +28,13 @@
 
 	RegisterSignal(parent, list(COMSIG_MOVABLE_MOVED,COMSIG_PARENT_PREQDELETED), .proc/disable)
 	RegisterSignal(parent, list(COMSIG_OBJ_DEFAULT_UNFASTEN_WRENCH), .proc/toggle_active)
+	RegisterSignal(parent, list(COMSIG_OBJ_HIDE), .proc/hide)
+	RegisterSignal(parent, list(COMSIG_ATOM_UPDATE_OVERLAYS), .proc/create_overlays) //create overlays also gets called after init (no idea by what it just happens)
 
 	if(start)
-		enable()
-
-	if(use_overlays)
-		create_overlays()
+		//timer 0 so it can finish returning initialize, after which we're added to the parent.
+		//Only then can we tell the duct next to us they can connect, because only then is the component really added. this was a fun one
+		addtimer(CALLBACK(src, .proc/enable), 0)
```

Comment:
can't you use LateInitialize hints for this? or are those not available on components?

Reply:
It wouldn't work here